storm-server/src/test/java/org/apache/storm/TestRebalance.java

```diff
@@ -151,28 +151,22 @@ public void waitTopologyScheduled(String topoName, ILocalCluster cluster, int re
 
     public boolean checkTopologyScheduled(String topoName, ILocalCluster cluster) throws TException {
         if (checkTopologyUp(topoName, cluster)) {
-            ClusterSummary sum = cluster.getClusterInfo();
-            for (TopologySummary topoSum : sum.get_topologies()) {
-                if (topoSum.get_name().equals(topoName)) {
-                    String status = topoSum.get_status();
-                    String sched_status = topoSum.get_sched_status();
-                    if (status.equals("ACTIVE") && (sched_status != null && !sched_status.equals(""))) {
-                        return true;
-                    }
-                }
+            TopologySummary topoSum = cluster.getTopologySummaryByName(topoName);
+            String status = topoSum.get_status();
+            String sched_status = topoSum.get_sched_status();
+            if (status.equals("ACTIVE") && (sched_status != null && !sched_status.equals(""))) {
+                return true;
             }
         }
         return false;
     }
 
     public boolean checkTopologyUp(String topoName, ILocalCluster cluster) throws TException {
         ClusterSummary sum = cluster.getClusterInfo();
-
-        for (TopologySummary topoSum : sum.get_topologies()) {
-            if (topoSum.get_name().equals(topoName)) {
+        TopologySummary topoSum = cluster.getTopologySummaryByName(topoName);
+            if (topoSum != null) {
```

Comment:
> Will it ever be `null`? Looks to me `getTopologySummaryByName` throws exception if it can't find the topo

Current implementation in the PR will throw exception if the topology can't be find (when calling `getTopologySummaryByName`). So we need to try-catch block to determine if this topology is up or not, instead of checking `topoSum!=null`.